### PR TITLE
[JENKINS-35661][BC_1_47] - When reading PCKS8 PrivateKey it should be possible to obtain a KeyPair, but toKeyPair is returnig null

### PR DESCRIPTION
**Pull request by alvarolobato**
[JENKINS-35661](https://issues.jenkins-ci.org/browse/JENKINS-35661)
In the case of getting a `RSAPrivateCrtKey` when decoding, it is possible to build a KeyPair instead of just returning the PrivateKey.
This is done by building a `PublicKeySpec` with the public exponent and the modulus from the private key and generating the public key with an RSA factory.

A test to check the issue was added.

**This is a backport of  #8  with some adaptations**

@reviewbybees

### CODE REVIEW

**Comment by amuniz** [comment]
Worth noting that this is the same patch than #8 for the 1.47 compatible branch.

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 1]
```diff
@@ -140,9 +144,23 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
...
+            if (object instanceof RSAPrivateCrtKey) {
+                // obtain public key spec from the private key
+                RSAPrivateCrtKey rsaPK = (RSAPrivateCrtKey) object;
+                RSAPublicKeySpec pubKeySpec = new RSAPublicKeySpec(rsaPK.getModulus(), rsaPK.getPublicExponent());
+                KeyFactory kf = KeyFactory.getInstance("RSA");
+                return new PEMEncodable(new KeyPair(kf.generatePublic(pubKeySpec), rsaPK));
+            }
+
+            return new PEMEncodable(object);
+        } catch (PasswordException | InvalidKeySpecException pwE) {
```
Comment (amuniz) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]: This branch has 1.609 as core dependency. IIRC that core version was supposed to support Java 6. So multi-catch syntax is not allowed.

BTW the `java.level` should be set to 6 in this branch.

Info [here](https://jenkins.io/blog/2015/04/06/good-bye-java6/).

Comment (amuniz) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]: I think this is a :bug:

Comment (alvarolobato) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]: Good catch, corrected.

**Comment by amuniz** [comment]
:bee:

**Comment by amuniz** [comment]
re-:bee:

**Comment by stephenc** [comment]
:bee:

**Comment by alvarolobato** [comment]
@reviewbybees done

**Pull request merged**